inst/include/plant/patch.h

```diff
@@ -176,12 +176,20 @@ void Patch<T,E>::rescale_environment() {
 
 template <typename T, typename E>
 void Patch<T,E>::compute_rates() {
-  for (size_t i = 0; i < size(); ++i) {
-    double pr_patch_survival = survival_weighting->pr_survival(time());
+  double pr_patch_survival = survival_weighting->pr_survival(time());
+  std::vector<std::vector<double>> resource_depletion; // n_species x n_layers
```

Comment:
Since size of this object is already known, could create and reserve size on patch creation